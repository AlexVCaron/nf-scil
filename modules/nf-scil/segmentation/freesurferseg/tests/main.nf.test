nextflow_process {

    name "Test Process SEGMENTATION_FREESURFERSEG"
    script "../main.nf"
    process "SEGMENTATION_FREESURFERSEG"

    tag "modules"
    tag "modules_nfcore"
    tag "segmentation"
    tag "segmentation/freesurferseg"

    test("segmentation - freesurferseg") {
        when {
            process {
                """
            input[0] = [
                [ id:'test', single_end:false ], // meta map
                file(params.test_data['segmentation']['freesurferseg']['aparc_aseg'], checkIfExists: true),
                file(params.test_data['segmentation']['freesurferseg']['wmparc'], checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
