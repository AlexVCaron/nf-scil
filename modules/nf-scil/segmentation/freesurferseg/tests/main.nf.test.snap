{
    "segmentation - freesurferseg": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "csf_mask": [
                    
                ],
                "gm_mask": [
                    
                ],
                "versions": [
                    
                ],
                "wm_mask": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-01T16:30:26.111498"
    }
}